src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.Build.Tasks.ResourceHandling
+{
+    internal class MSBuildResXReader
+    {
+        public IReadOnlyList<IResource> Resources { get; }
+
+        public static IReadOnlyList<IResource> ReadResources(Stream s, string filename, bool pathsRelativeToBasePath)
+        {
+            var resources = new List<IResource>();
+            var aliases = new Dictionary<string, string>();
+
+            using (var xmlReader = new XmlTextReader(s))
+            {
+                xmlReader.WhitespaceHandling = WhitespaceHandling.None;
+
+                XDocument doc = XDocument.Load(xmlReader, LoadOptions.PreserveWhitespace);
+                foreach (XElement elem in doc.Element("root").Elements())
+                {
+                    switch (elem.Name.LocalName)
+                    {
+                        case "schema":
+                            // TODO: this
+                            break;
+                        case "assembly":
+                            ParseAssemblyAlias(aliases, elem);
+                            break;
+                        case "resheader":
+                            break;
+                        case "data":
+                            ParseData(filename, pathsRelativeToBasePath, resources, aliases, elem);
+                            break;
+                        default:
+                            throw new NotImplementedException();
+                    }
+                }
+            }
+
+            return resources;
+        }
+
+        private static void ParseAssemblyAlias(Dictionary<string,string> aliases, XElement elem)
+        {
+            string alias = elem.Attribute("alias")?.Value;
+            string name = elem.Attribute("name").Value;
+
+            if (string.IsNullOrEmpty(alias))
+            {
+                AssemblyName assemblyName = new AssemblyName(name);
```

Comment:
@ericstj is this safe, or might it result in loading an assembly? It's what ResXResourceReader did:

https://github.com/dotnet/winforms/blob/33b9fe202f3dc1b8e7c4bf28492f8bd3252f1a20/src/System.Windows.Forms/src/System/Resources/ResXResourceReader.cs#L575-L587

Reply:
Yeah. AssemblyName is safe it’s just a parsed string.